python/ray/worker.py

```diff
@@ -931,16 +931,33 @@ def _process_task(self, task, function_execution_info):
         finally:
             self._current_task = None
 
-        # Store the outputs in the local object store.
+        retries_left = 5
+        delay = 1
         try:
-            with profiling.profile("task:store_outputs"):
-                # If this is an actor task, then the last object ID returned by
-                # the task is a dummy output, not returned by the function
-                # itself. Decrement to get the correct number of return values.
-                num_returns = len(return_object_ids)
-                if num_returns == 1:
-                    outputs = (outputs, )
-                self._store_outputs_in_object_store(return_object_ids, outputs)
+            while retries_left:
```

Comment:
I think we should actually put this retry loop in a lower-level function, either `Worker.put_object` or `Worker.store_and_register`. The current fix might work for the return value problem, but we'll probably run into the same issue with `ray.put`. Also, it seems like this might not work if there are multiple return values and only one of them fails, since then we'll try to put all of the return values into the store again.